Mathlib/MeasureTheory/Measure/Tight.lean

```diff
@@ -0,0 +1,469 @@
+    (hKclosed : ∀ n, IsClosed (K n)) (h : μ (⋃ n, K n) = μ Set.univ) :
+    ∀ ε > 0, ∃ n, μ ((Set.Accumulate K n)ᶜ) ≤ ε := by
+  rintro ε hε
+  obtain ⟨n, hn⟩ := almost_cover_has_approx_accumulate K h ε hε
+  use n
+  have hK2 : IsClosed (Set.Accumulate K n) := by
+    apply Set.Finite.isClosed_biUnion ?_ (fun i _ => hKclosed i)
+    apply Set.Finite.to_subtype
+    convert (Finset.Iic n).finite_toSet using 1
+    simp only [Nat.le_eq, Finset.coe_Iic]
+    rfl
+  rw [measure_compl hK2.measurableSet (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hn
+
+/-- For a finite measure `μ`, we can construct a dense sequence such that for any radius, we can
+cover all but a set of measure below `ε`. -/
+private lemma approx_ball_cover_of_separableSpace [MeasurableSpace α] [PseudoMetricSpace α]
+    [TopologicalSpace.SeparableSpace α] [Nonempty α] (μ : Measure α) [IsFiniteMeasure μ] :
+    ∃ K : ℕ → α, DenseRange K ∧
+    ∀ ε > 0, ∀ δ > 0, ∃ N : ℕ, μ (Set.univ) - ε ≤ μ (⋃ i ≤ N, Metric.ball (K i) δ)  := by
+  obtain ⟨K, hK⟩ := TopologicalSpace.exists_dense_seq α
+  use K, hK
+  intro ε hε δ hδ
+  apply almost_cover_has_approx_accumulate (fun y ↦ Metric.ball (K y) δ) (?_) ε hε
+  apply le_antisymm (measure_mono fun ⦃a⦄ _ ↦ trivial)
+  exact measure_mono <| fun y _ => Set.mem_iUnion.mpr (DenseRange.exists_dist_lt hK y hδ)
+
+/-- For a finite measure `μ`, we can construct a dense sequence such that for any radius, we can
+find a `N` such that the measure of the complement of the first `N` balls of radius `1/(j+1)` is
+at most `ε/2^(j+1)`. -/
+private lemma approx_ball_cover_of_separableSpace_compl [MeasurableSpace α] [PseudoMetricSpace α]
+    [TopologicalSpace.SeparableSpace α] [Nonempty α] [OpensMeasurableSpace α] (μ : Measure α)
+    [IsFiniteMeasure μ] {ε : ENNReal} (hε : 0 < ε) : ∃ K : ℕ → α, DenseRange K ∧
+    ∀ j : ℕ, ∃ N : ℕ, μ ((⋃ i ≤ N, Metric.ball (K i) (1/(j+1)))ᶜ) ≤ ε/2^(j+1) := by
+  obtain ⟨K, hK, cover⟩ := approx_ball_cover_of_separableSpace μ
+  have hεj_pos : ∀ j : ℕ, ε/(2^(j+1)) > 0 :=
+    fun j => ENNReal.div_pos hε.ne' (Ne.symm (ne_of_beq_false rfl))
+  use K, hK
+  intro j
+  obtain ⟨N, hN⟩ := cover (ε/(2^(j+1))) (hεj_pos j) (1/(j+1)) (Nat.one_div_pos_of_nat)
+  use N
+  rw [measure_compl (by measurability) (measure_ne_top μ _)]
+  exact tsub_le_iff_tsub_le.mp hN
+
+/-- A measure `μ` is separable if there is a separable set `S` such that `μ S = μ Set.univ`. -/
+ def IsSeparable [TopologicalSpace α] (μ : Measure α) : Prop :=
+   ∃ S : Set α, TopologicalSpace.IsSeparable S ∧ μ S = μ Set.univ
+
+namespace IsSeparable
+
+lemma of_separableSpace [TopologicalSpace α] [TopologicalSpace.SeparableSpace α] :
+    IsSeparable μ := ⟨Set.univ, TopologicalSpace.isSeparable_univ_iff.mpr ‹_›, rfl⟩
+
+end IsSeparable
+
+/-- A measure `μ` is pre-tight if for all `0 < ε`, there exists `K` totally bounded such that
+  `μ Kᶜ ≤ ε`. -/
+def IsPretight [UniformSpace α] (μ : Measure α) : Prop :=
```

Comment:
You are going to have a standing assumption `UniformSpace α` in a lot of lemmas below. It's probably worth putting it on a `variable` line (opening a section to contain its scope if necessary).